src/lib/containers/DirectDownloadButton.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect } from 'react'
+import { Button } from 'react-bootstrap'
+import { BatchFileRequest, FileHandleAssociation } from '../utils/synapseTypes'
+import { getFiles } from '../utils/SynapseClient'
+
+export type DirectFileDownloadButtonProps = {
+  id?: string
+  token: string
+  fileHandleAssociation: FileHandleAssociation
+  fileName: string | undefined
+  className?: string
+  variant?: string  // This allows you to change the look of the button (see react bootstrap doc)
+}
+
+const DirectDownloadButton: React.FC<DirectFileDownloadButtonProps> = props => {
+
+  const {id, token, fileHandleAssociation, className, variant, fileName}  = props
+  let mounted:boolean = true
+
+  useEffect( () => {
+    if (mounted) {
+      //
+    }
+    return () => {
+      mounted = false
+    }
+  }, [token])
+
+  const getDownloadLink = async () => {
+    if (!fileHandleAssociation.fileHandleId || !token) return
+
+    const batchFileRequest:BatchFileRequest = {
+      requestedFiles: [fileHandleAssociation],
+      includePreSignedURLs: true,
+      includeFileHandles: true,
```

Comment:
I think includeFileHandles can be false in this case.

Reply:
Done.